test/jobs/job-status-pagination.ts

```diff
@@ -0,0 +1,209 @@
+      });
+    });
+
+    describe('and the limit is invalid', function () {
+      hookJobStatus({ jobID, username: 'joe', query: { limit: 10001 } });
+      it('returns a 400 HTTP Bad request response', function () {
+        expect(this.res.statusCode).to.equal(400);
+      });
+
+      it('returns a JSON error response', function () {
+        const response = JSON.parse(this.res.text);
+        expect(response).to.eql({
+          code: 'harmony.RequestValidationError',
+          description: 'Error: Parameter "limit" is invalid. Must be an integer greater than or equal to 0 and less than or equal to 10000.',
+        });
+      });
+    });
+
+    describe('and the limit is not an integer', function () {
+      hookJobStatus({ jobID, username: 'joe', query: { limit: 2000.5 } });
+      it('returns a 400 HTTP Bad request response', function () {
+        expect(this.res.statusCode).to.equal(400);
+      });
+
+      it('returns a JSON error response', function () {
+        const response = JSON.parse(this.res.text);
+        expect(response).to.eql({
+          code: 'harmony.RequestValidationError',
+          description: 'Error: Parameter "limit" is invalid. Must be an integer greater than or equal to 0 and less than or equal to 10000.',
+        });
+      });
+    });
+  });
+
+  describe('when both `page` and `limit` parameters are set', function () {
+    hookJobStatus({ jobID, username: 'joe', query: { page: 2, limit: 2 } });
+    it('shows the corresponding page of results with the correct page size', function () {
+      const job = new Job(JSON.parse(this.res.text));
+      const outputLinks = job.getRelatedLinks('data');
+      expect(areJobLinksEqual(links.slice(2, 4), outputLinks)).to.equal(true);
+    });
+  });
+
+  describe('link relations', function () {
+    describe('on the first page', function () {
+      hookJobStatus({ jobID, username: 'joe', query: { page: 1 } });
+      it('does not provide a "prev" link relation to the previous result page', function () {
+        const job = new Job(JSON.parse(this.res.text));
+        const relLinks = job.getRelatedLinks('prev');
+        expect(relLinks.length).to.equal(0);
+      });
+
+      it('provides a "next" link relation to the previous result page', function () {
+        const job = new Job(JSON.parse(this.res.text));
+        const relLinks = job.getRelatedLinks('next');
+        expect(relLinks.length).to.equal(1);
+      });
+    });
+
+    describe('on the second page', function () {
+      hookJobStatus({ jobID, username: 'joe', query: { page: 2 } });
+      itIncludesPagingRelations(5, `jobs/${jobID}`, { first: null, prev: 1, self: 2, next: 3, last: 5 });
+    });
+
+    describe('on a middle page', function () {
+      hookJobStatus({ jobID, username: 'joe', query: { page: 3 } });
+      itIncludesPagingRelations(5, `jobs/${jobID}`, { first: 1, prev: 2, self: 3, next: 4, last: 5 });
+    });
+
+    describe('on the penultimate page', function () {
```

Comment:
Huh so second to last is what penultimate means?

Reply:
yes